addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -545,6 +553,49 @@ var SnippetEditor = Widget.extend({
         ev.preventDefault();
         this.clone(true);
     },
+    /**
+     * Called while snippet is dragged.
+     *
+     * @private
+     */
+    _onDrag: function (ev, ui) {
+        const draggableOffset = this._draggableCursorAt.top + ui.offset.top;
+        const scrollOffsetThreshold = 100;
+        const scrollStep = 10;
+        const $scrollTarget = $('html, body');
+        const scrollTimer = 10;
+        const visibleBottomOffset = $scrollTarget.scrollTop() +
+                                    $scrollTarget.get(0).clientHeight -
+                                    draggableOffset;
+        const visibleTopOffset = draggableOffset - $scrollTarget.scrollTop();
+
+        if (visibleTopOffset <= scrollOffsetThreshold + this._mainNavBarHeight) {
+            this._clearAutoScrollInterval();
+            this._autoScrollHandler = setInterval(function () {
+                    $scrollTarget.scrollTop($scrollTarget.scrollTop() -
+                                            scrollStep *
+                                            (
+                                                1 - visibleTopOffset /
+                                                (scrollOffsetThreshold + this._mainNavBarHeight)
+                                            ));
+                }.bind(this),
+                scrollTimer
+            );
```

Comment:
Indentation + ES6
```suggestion
            this._autoScrollHandler = setInterval(() => {
                $scrollTarget.scrollTop($scrollTarget.scrollTop() -
                                        scrollStep *
                                        (
                                            1 - visibleTopOffset /
                                            (scrollOffsetThreshold + this._mainNavBarHeight)
                                        ));
            }, scrollTimer);
```
or
```suggestion
            this._autoScrollHandler = setInterval(
                () => {
                    $scrollTarget.scrollTop($scrollTarget.scrollTop() -
                                            scrollStep *
                                            (
                                                1 - visibleTopOffset /
                                                (scrollOffsetThreshold + this._mainNavBarHeight)
                                            ));
                },
                scrollTimer
            );
```

Reply:
Done!